nixos/modules/services/logging/awstats.nix

```diff
@@ -4,31 +4,130 @@ with lib;
+        example = "mail";
+        description = ''
+          The type of log being collected.
+        '';
+      };
+      domain = mkOption {
+        type = types.str;
+        default = name;
+        description = "The domain name to collect stats for";
+        example = "example.com";
+      };
+
+      logFile = mkOption {
+        type = types.str;
+        example = "/var/spool/nginx/logs/access.log";
+        description = ''
+          The log file to be scanned.
+
+          For mail, set this to
+          <literal>
+          journalctl $OLD_CURSOR -u postfix.service | ${pkgs.perl}/bin/perl ${package.out}/share/awstats/tools/maillogconvert.pl standard |
+          </literal>
+        '';
+      };
+
+      logFormat = mkOption {
+        type = types.str;
+        default = "1";
+        description = ''
+          The log format being used.
 
+          For mail, set this to
+          <literal>
+          %time2 %email %email_r %host %host_r %method %url %code %bytesd
+          </literal>
+        '';
+      };
+
+      hostAliases = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        example = "[ \"www.example.org\" ]";
+        description = ''
+          List of aliases the site has.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.attrsOf types.str;
+        default = {};
+        example = literalExample ''
+          {
+            "ValidHTTPCodes" = "404";
+          }
+        '';
+      };
+
+      webService = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Enable the awstats web service";
+        };
+
+        hostname = mkOption {
+          type = types.str;
+          default = config.domain;
+          description = "The hostname the web service appears under";
+        };
+
+        urlPrefix = mkOption {
+          type = types.str;
+          default = "/awstats";
+          description = "The URL prefix under which the awstats pages appear.";
+        };
+      };
+    };
+  };
+  webServices = filterAttrs (name: value: value.webService.enable) cfg.configs;
+in
 {
   options.services.awstats = {
     enable = mkOption {
       type = types.bool;
-      default = cfg.service.enable;
+      default = false;
       description = ''
         Enable the awstats program (but not service).
         Currently only simple httpd (Apache) configs are supported,
         and awstats plugins may not work correctly.
       '';
     };
-    vardir = mkOption {
+
+    dataDir = mkOption {
       type = types.path;
       default = "/var/lib/awstats";
-      description = "The directory where variable awstats data will be stored.";
+      description = "The directory where awstats data will be stored.";
     };
 
-    extraConfig = mkOption {
-      type = types.lines;
-      default = "";
-      description = "Extra configuration to be appendend to awstats.conf.";
+    runDir = mkOption {
```

Comment:
Is there a specific need to allow the user to customize the runtime directory? If this directory simply holds transient files I would suggest removing the `runDir` option in favor of the `systemd` `RuntimeDirectory` option, if possible.

Reply:
The web stats don't need a runtime directory - this is just for the mail. It stores a cursor for where the logs were last analysed up to.

I don't see why that can't go in /var, so I moved it there. @griff do you have any comments?

Reply:
@griff is there any feedback you could offer here?

Reply:
I noticed that `runDir` isn't even being used, so this option can be removed.